editor/src/components/canvas/canvas-strategies/post-action-options/navigator-reparent.ts

```diff
@@ -91,10 +92,35 @@ function getNavigatorReparentCommands(
       elementPath: path,
       pathToReparent: pathToReparent(path),
       intendedCoordinates: intendedCoordinates,
-      uid: EP.toUid(path),
+      newUID: EP.toUid(path),
     }
   })
 
+  const oldPathToNewPathMapping: OldPathToNewPathMapping = data.dragSources.reduce(
+    (acc: OldPathToNewPathMapping, value) => {
+      const childrenPaths = MetadataUtils.getChildrenPathsOrdered(
+        editor.jsxMetadata,
+        editor.elementPathTree,
+        value,
+      )
+
+      const pathAfterReparent = elementPathFromInsertionPath(newParentPath, EP.toUid(value))
+
+      const mapping: OldPathToNewPathMapping = [value, ...childrenPaths].reduce(
+        (accI: OldPathToNewPathMapping, valI) => ({
+          ...accI,
+          [EP.toString(valI)]: EP.replaceIfAncestor(valI, value, pathAfterReparent) ?? undefined,
+        }),
+        {},
+      )
+      return {
```

Comment:
this is probably not on the performance hot path, but it's important to mutate the accumulator as it otherwise creates a ton of work for the browser to copy the objects and immediately garbage collect them!

Reply:
This has two nested reduces, with the inner one fed by a spread array too. Turning those into nested for...of loops, with the inner one looping over `childrenPaths` after running the same code for `value` would be a huge speed bump to this.

Reply:
Berci you may want to try our new invention, `arrayAccumulate` ! :)

Reply:
I refactored the nested `reduce`s to nested for loops in [0f76da3](https://github.com/concrete-utopia/utopia/pull/3890/commits/0f76da3a6a5188a7f08856845393ba804c6d4786)